app/controllers/alchemy/pages_controller.rb

```diff
@@ -39,13 +39,12 @@ class PagesController < Alchemy::BaseController
     #
     # If no public page can be found it renders a 404 error.
     #
-    # If the configuration is set to :redirect_index, then the request gets redirected
-    # to that page, instead of displaying it.
-    #
     def index
       @page || page_not_found!
 
       if Alchemy::Config.get(:redirect_index)
+        ActiveSupport::Deprecation.warn("The configuration option `redirect_index` is deprecated and will be removed with the release of Alchemy v4.0")
```

Comment:
Metrics/LineLength: Line is too long. [151/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)